src/components/Api.js

```diff
@@ -0,0 +1,114 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+    this._authorization = headers.authorization;
+    this._contentType = headers["Content-Type"];
+  }
+
+  //Check if JSON object ok
+  _checkResponseOk(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      Promise.reject(`Erorr: ${res.status}`)
+    }
+  }
+
+  //Get initial cards from URL
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    }).then(this._checkResponseOk)
```

Comment:
**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponseOk)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.